src/software/embedded/primitive_executor.cpp

```diff
@@ -114,12 +114,19 @@ std::unique_ptr<TbotsProto::DirectControlPrimitive> PrimitiveExecutor::stepPrimi
             auto output = std::make_unique<TbotsProto::DirectControlPrimitive>(
                 prim->direct_control());
             status.set_running_primitive(false);
+            if (output->motor_control().has_direct_velocity_control()){
```

Comment:
instead of duplicating this logic across all the different switch cases, you can have added this in the `Thunderloop` poll function after the `stepPrimitive` function